app/models/patient_session.rb

```diff
@@ -126,4 +128,23 @@ def latest_triage
   def consents_to_send_communication
     latest_consents.select(&:response_given?).reject(&:via_self_consent?)
   end
+
+  def pending_transfer?
+    proposed_session_id.present?
+  end
+
+  def confirm_transfer!
+    return unless pending_transfer?
+
+    if vaccination_records.any?
```

Comment:
We also need to consider gillick assessments here, we should use `!safe_to_destroy?` I think.